integration_tests/suite/test_phonebook_http.py

```diff
@@ -358,3 +391,304 @@ def test_get_multi_tenant(self, sub, main):
             self.assert_unknown_source_exception(main['uuid'], e)
         else:
             self.fail('Should have raised')
+
+
+class TestGetContacts(BasePhonebookCRUDTestCase):
+    @classmethod
+    def setUpClass(cls) -> None:
+        super().setUpClass()
+        cls.phonebook_crud = PhonebookCRUD(cls.Session)
+        cls.contact_crud = PhonebookContactCRUD(cls.Session)
+        cls.stack = ExitStack()
+
+        cls.phonebook = cls.phonebook_crud.create(
+            MAIN_TENANT,
+            {'name': 'test-phonebook', 'description': 'some test phonebook'},
+        )
+        cls.stack.callback(
+            cls.phonebook_crud.delete, None, PhonebookKey(uuid=cls.phonebook['uuid'])
+        )
+        cls.contacts, errors = cls.contact_crud.create_many(
+            [MAIN_TENANT],
+            PhonebookKey(uuid=cls.phonebook['uuid']),
+            [
+                {
+                    'firstname': f'Contact {i}',
+                    'lastname': 'McContact',
+                    'number': str(1000000000 + i),
+                }
+                for i in range(5000)
+            ],
+        )
+        assert not errors
+
+    @classmethod
+    def tearDownClass(self):
+        self.stack.close()
+        super().tearDownClass()
+
+    def test_get_all(self):
+        client = self.get_client(VALID_TOKEN_MAIN_TENANT)
+
+        with self.source(
+            client,
+            {
+                'phonebook_uuid': self.phonebook['uuid'],
+                'name': self.phonebook['name'] + "-source",
+            },
+        ) as source:
+            response = client.session().get(
+                url=f"{client.phonebook_source.base_url}/{source['uuid']}/contacts"
+            )
+            response.raise_for_status()
+            body = response.json()
+            assert_that(
+                body,
+                has_entries(
+                    total=len(self.contacts),
+                    filtered=len(self.contacts),
+                    items=all_of(
+                        has_length(len(self.contacts)),
+                        only_contains(
+                            has_entries(
+                                firstname=instance_of(str),
+                                lastname=instance_of(str),
+                                number=instance_of(str),
+                                id=instance_of(str),
```

Comment:
This does not assert that we have all contacts: the test would pass if the listing returned 5000 times the same contact.